File: apina-core/src/main/java/fi/evident/apina/spring/SpringAnnotationUtils.java
package fi.evident.apina.spring;

import fi.evident.apina.java.model.JavaAnnotation;

import java.util.Optional;

final class SpringAnnotationUtils {
    static Optional<String> getRequestParamName(JavaAnnotation annotation) {
        return valueOrSecondaryAttribute(annotation, "name");
    }

    public static Optional<String> getRequestMappingPath(JavaAnnotation annotation) {
        return valueOrSecondaryAttribute(annotation, "path");
    }

    private static Optional<String> valueOrSecondaryAttribute(JavaAnnotation annotation, String secondaryName) {
        Optional<String> value = annotation.getAttribute("value", String.class);
        if (value.isPresent())
            return value;
        else
            return annotation.getAttribute(secondaryName, String.class);
    }
}

File: apina-core/src/main/kotlin/fi/evident/apina/spring/SpringAnnotationUtils.kt
package fi.evident.apina.spring

import fi.evident.apina.java.model.JavaAnnotation

internal object SpringAnnotationUtils {
    fun getRequestParamName(annotation: JavaAnnotation): String? =
        valueOrSecondaryAttribute(annotation, "name")

    fun getRequestMappingPath(annotation: JavaAnnotation): String? =
        valueOrSecondaryAttribute(annotation, "path")

    private fun valueOrSecondaryAttribute(annotation: JavaAnnotation, secondaryName: String): String? =
        annotation.getAttribute<String>("value") ?: annotation.getAttribute<String>(secondaryName)
}
